File: TennisLeague.Web/ClientApp/src/models/LeagueSeason.ts
import { addDaysToDate } from "../utilities/dateTimeHelpers";

export class LeagueSeason {
    constructor() {
        this.seasonID = 1;
        this.year = new Date().getFullYear();
        this.startDate = new Date(new Date().toDateString());
        this.durationInWeeks = 9;
        this.statusID = 1;
        this.registrationDate = addDaysToDate(this.startDate, -14);
    }

    id?: number;
    seasonID: number;
    year: number;
    startDate: Date;
    durationInWeeks: number;
    statusID: number;
    registrationDate: Date;
}

File: TennisLeague.API/Models/LeagueSeason.cs
namespace TennisLeague.API.Models
{
    public class LeagueSeason
    {
        public int? ID { get; set; }
        public int SeasonID { get; set; }
        public int Year { get; set; }
        public DateTime StartDate { get; set; }
        public int DurationInWeeks { get; set; }
        public int StatusID { get; set; }
        public DateTime RegistrationDate { get; set; }
    }
}
